File: data_structures/q7/solution.py
def secondLargest(arr):
    if(len(arr) < 2): 
        return -1
    largest = float('-inf')
    second = float('-inf')
    num = 0
    
    for i in arr:
        if(i > largest):
            second = largest
            largest = i
        
        elif(largest > num > second):
            second = num
    if(second == float('inf')):
        return -1
    return second

arr = [1,2,4,5]
print("Second largest nnumber is: ",secondLargest(arr))

File: data_structures/q7/solution.cpp
#include<iostream>
#include<climits>
#include<vector>
using namespace std;

int secondLargest(vector<int> arr, int n){

    if (n < 2){
        return -1;
    }
    int largest = INT_MIN;
    int second = INT_MIN;

    for(int i = 0; i < n; i++ ){
        if(arr[i] > largest){
            second = largest;
            largest = arr[i];
        }
        else if(arr[i] > second && arr[i] != largest){
            second = arr[i];
        }
    }
    if (second == INT_MIN){
        return -1; 
    }
    
    return second;
}

int main(){
    vector<int> arr = {1,23,4,5};
    int n = arr.size();

    cout << "The second largest integer is: " << secondLargest(arr,n) << endl;
    return 0;

}